channeldb/channel_test.go

```diff
@@ -98,21 +109,141 @@ func makeTestDB() (*DB, func(), error) {
 	return cdb, cleanUp, nil
 }
 
-func createTestChannelState(cdb *DB) (*OpenChannel, error) {
+// testChannelParams is a struct which details the specifics of how a channel
+// should be created.
+type testChannelParams struct {
+	// channel is the channel that will be written to disk.
+	channel *OpenChannel
+
+	// addr is the address that the channel will be synced pending with.
+	addr *net.TCPAddr
+
+	// pendingHeight is the height that the channel should be recorded as
+	// pending.
+	pendingHeight uint32
+
+	// openChannel is set to true if the channel should be fully marked as
+	// open if this is false, the channel will be left in pending state.
+	openChannel bool
+}
+
+// testChannelOption is a functional option which can be used to alter the
+// default channel that is creates for testing.
+type testChannelOption func(params *testChannelParams)
+
+// pendingHeightOption is an option which can be used to set the height the
+// channel is marked as pending at.
+var pendingHeightOption = func(height uint32) testChannelOption {
```

Comment:
Is there any advantage of using global variables vs declaring actual functions?

Reply:
Just a design pattern I've picked up somewhere :) functions definitely better than global vars.

Reply:
:+1: :slightly_smiling_face:

Reply:
Maybe convert them to functions to maintain consistency with the rest of the project? Not a blocker though.